lib/trip.rb

```diff
@@ -1,33 +1,55 @@
-require 'csv'
+require "csv"
+require "time"
 
-require_relative 'csv_record'
+require_relative "csv_record"
 
 module RideShare
   class Trip < CsvRecord
-    attr_reader :id, :passenger, :passenger_id, :start_time, :end_time, :cost, :rating
+    attr_reader :id, :passenger, :passenger_id, :start_time, :end_time, :cost, :rating, :driver_id, :driver
 
-    def initialize(id:,
-      passenger: nil, passenger_id: nil,
-      start_time:, end_time:, cost: nil, rating:)
+    def initialize(id:, driver: nil, driver_id: nil, passenger: nil, passenger_id: nil, start_time:, end_time:, cost: nil, rating:)
       super(id)
 
+      # @driver = driver
+      # @driver_id = driver_id
+      if driver
+        @driver = driver
+        @driver_id = driver.id
+      elsif driver_id
+        @driver_id = driver_id
+      else
+        raise ArgumentError, "Driver or driver ID is required"
+      end
+
       if passenger
         @passenger = passenger
         @passenger_id = passenger.id
-
       elsif passenger_id
         @passenger_id = passenger_id
+      else
+        raise ArgumentError, "Passenger or passenger_id is required"
+      end
 
+      if start_time == nil
```

Comment:
This if statement doesn't make sense.  You're setting start_time to nil if start_time is nil

This would be better as:


```ruby
@start_time = Time.parse(start_time) unless start_time.nil?
```